### PR TITLE
Stripe connect

### PR DESCRIPTION
**Pull request by jasonfiammetta**
### What this PR does (required):
Adds the ability for a PetSitter account to create a Stripe Connect account linked to their user account
Closes Issue #83 

### Screenshots / Videos (front-end only):
Screen Shot 2022-01-31 at 12 19 50 PM

### Any information needed to test this feature (required):
- Login, go to settings, payment methods, and click the button that says "Create a Stripe Account"

### Any issues with the current functionality (optional):
- Currently there is no way to get a PetSitter account
- Ensure that your .env file has been updated with the Stripe keys

### CODE REVIEW

File: server/controllers/stripe.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          success: {
+            stripeAccount,
+          },
+        });
+      } else {
+        res.status(500).json({ error: "Unable to connect to Stripe API" });
+      }
+    })
+    .catch((error) => {
+      console.log(error);
```
Comment (sundayezeilo) [inline server/controllers/stripe.js]: Let's use `console.error` for this one.

File: server/controllers/stripe.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      });
+    })
+    .then((stripeAccount) => {
+      if (stripeAccount.url) {
+        res.json({
+          success: {
+            stripeAccount,
+          },
+        });
+      } else {
```
Comment (sundayezeilo) [inline server/controllers/stripe.js]: Let's make this code cleaner and more readable. Prefer to use `async`/`await`.

File: server/controllers/stripe.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
+const asyncHandler = require("express-async-handler");
+const PetSitter = require("../models/PetSitter");
+const stripe = require("stripe")(process.env.STRIPE_SECRET);
+
+// @route POST /connect/stripe
+// @desc redirect to Stripe onboarding
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/stripe.js]: Good documentation

File: client/src/helpers/APICalls/stripe.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  const fetchOptions: FetchOptions = {
+    method: 'POST',
+    credentials: 'include',
+  };
+  return await fetch(`/connect/stripe`, fetchOptions)
+    .then((res) => res.json())
+    .catch(() => ({
+      error: { message: 'Unable to connect to server. Please try again' },
+    }));
+};
```
Comment (sundayezeilo) [inline client/src/helpers/APICalls/stripe.ts]: Using `then()`/`catch()` makes `async`/await` redundant. Please use `async`/`await` properly.

**Review by sundayezeilo — CHANGES_REQUESTED** [review]
Looks good. Please address the comments below.
Happy coding!

File: client/src/helpers/APICalls/stripe.ts (hunk 4 above)
Comment (jasonfiammetta) [reply client/src/helpers/APICalls/stripe.ts]: Currently every file in the ```APICalls``` folder has this same redundancy. I'll fix it in ```APICalls/stripe.ts``` for this PR, but the others should be brought up in their own issue so the rest of the app can be tested again.

**Review by moffatethan — APPROVED** [review]

**Pull request merged**